### ISSUE TITLE
Named scheduled tasks don't respect HazelcastInstanceAware marker

### ISSUE THREAD

**Issue by tkountis** [hazelcast/hazelcast#15230]
Plain runnable

```
public class X implements Runnable, Serializable, HazelcastInstanceAware
{

    private transient HazelcastInstance hz;

    @Override
    public void run()
    {
            // do sth with hz
    }

    @Override
    public void setHazelcastInstance(HazelcastInstance hazelcastInstance)
    {
        this.hz = hazelcastInstance;
    }
}
```

When scheduled with a name:

```
IScheduledExecutorService executor = hzInstance.getScheduledExecutorService("default");
executor.scheduleAtFixedRate(TaskUtils.named("X_Name", new X()), 5, 10, TimeUnit.SECONDS);
```
it doesn't get an instance associated with it.

**Comment by mmedenjak** [comment]
Usually the serialization service injects any context if it's used to deserialise the object. I'm guessing the task here was executed locally and not serialised?

**Comment by mmedenjak** [comment]
@tkountis can you make a forward-port so we close this PR?

**Comment by mmedenjak** [comment]
Closing as there is now a 4.0 PR to remind us to fix it in master.

**Comment by afspear** [comment]
Any chance this will get merged down to 3.x?

**Comment by tkountis** [comment]
It’s already merged for 3.12.2
https://github.com/hazelcast/hazelcast/pull/15233
